receiver/hostmetricsreceiver/internal/scraper.go

```diff
@@ -22,15 +22,26 @@ import (
 	"go.opentelemetry.io/collector/consumer/pdata"
 )
 
-// Scraper gathers metrics from the host machine and converts
-// these into internal metrics format.
-type Scraper interface {
+// BaseScraper gathers metrics from the host machine.
+type BaseScraper interface {
 	// Initialize performs any timely initialization tasks such as
 	// setting up performance counters for initial collection.
 	Initialize(ctx context.Context) error
 	// Close should clean up any unmanaged resources such as
 	// performance counter handles.
 	Close(ctx context.Context) error
+}
+
+// BaseFactory can create a Scraper.
+type BaseFactory interface {
+	// CreateDefaultConfig creates the default configuration for the Scraper.
+	CreateDefaultConfig() Config
+}
+
+// Scraper gathers metrics from the host machine.
+type Scraper interface {
```

Comment:
Yea I agree this name is definitely a bit confusing. I'm not sure what to call it though, as I'd prefer to leave "Host" out of the name since it's not an explicit requirement that metrics are scraped from the host - would something like `MetricsSliceScraper` & `ResourceMetricsSliceScraper` be better?

I'll create a separate PR to do this refactor as it will impact all the other scrapers.